tests/integration_tests/test_lm/test_mpirun.py

```diff
@@ -16,27 +16,30 @@ class TestTask(TestCase):
     # --------------------------------------------------------------------------
     #
     def setUp(self) -> dict:
-        path = os.path.dirname(__file__) + '../test_config/resources.json'
+        path = os.path.dirname(__file__) + '/../test_config/resources.json'
         resources = ru.read_json(path)
         hostname = socket.gethostname()
 
         for host in resources.keys():
             if host in hostname:
-                return resources[host]
+                return resources[host], host
 
     # --------------------------------------------------------------------------
     #
     @mock.patch.object(MPIRun, '__init__',   return_value=None)
-    def test_configure(self, mocked_init):
-        cfg = self.setUp()
+    @mock.patch('radical.utils.Logger')
+    def test_configure(self, mocked_init, mocked_Logger):
+        cfg, host = self.setUp()
         component = MPIRun(name=None, cfg=None, session=None)
-        component._log = ru.Logger('dummy')
-        component._cfg = {}
+        component.name = 'mpirun'
+        component._log = mocked_Logger
+        component._cfg = ru.Munch({'resource': host})
         component.env_removables = []
         component._configure()
+
         self.assertEqual(component.launch_command, cfg['mpirun_path'])
-        self.assertEqual(component.mpi_flavor, cfg['mpi_flavor'])
         self.assertEqual(component.mpi_version, cfg['mpi_version'])
+        self.assertEqual(component.mpi_flavor, cfg['mpi_flavor'])
```

Comment:
would be better to have `cfg.get(...)`, thus if resource config doesn't have that key, we wouldn't have `KeyError`, but rather `AssertError`.

(*) here `cfg` is `dict`, but if it would be `Munch`/`Config`-based then it would work fine without method `get()`

Reply:
Not so sure. `cfg` here is the test config and not the resource config. It may be better if I name it `results` or something. If an entry is missing means that the test is missing a test case.